src/shared/conf-parser.h

```diff
@@ -121,44 +121,45 @@ int config_parse_many(
+#define GENERIC_PARSER_ARGS \
+                const char *unit,                                       \
+                const char *filename,                                   \
+                unsigned line,                                          \
+                const char *section,                                    \
+                unsigned section_line,                                  \
+                const char *lvalue,                                     \
+                int ltype,                                              \
+                const char *rvalue,                                     \
+                void *data,                                             \
+                void *userdata
+int config_parse_int(GENERIC_PARSER_ARGS);
+int config_parse_unsigned(GENERIC_PARSER_ARGS);
+int config_parse_long(GENERIC_PARSER_ARGS);
+int config_parse_uint8(GENERIC_PARSER_ARGS);
+int config_parse_uint16(GENERIC_PARSER_ARGS);
+int config_parse_uint32(GENERIC_PARSER_ARGS);
+int config_parse_uint64(GENERIC_PARSER_ARGS);
+int config_parse_double(GENERIC_PARSER_ARGS);
+int config_parse_iec_size(GENERIC_PARSER_ARGS);
+int config_parse_si_size(GENERIC_PARSER_ARGS);
+int config_parse_iec_uint64(GENERIC_PARSER_ARGS);
+int config_parse_bool(GENERIC_PARSER_ARGS);
+int config_parse_tristate(GENERIC_PARSER_ARGS);
+int config_parse_string(GENERIC_PARSER_ARGS);
+int config_parse_path(GENERIC_PARSER_ARGS);
+int config_parse_strv(GENERIC_PARSER_ARGS);
+int config_parse_sec(GENERIC_PARSER_ARGS);
+int config_parse_nsec(GENERIC_PARSER_ARGS);
+int config_parse_mode(GENERIC_PARSER_ARGS);
+int config_parse_log_facility(GENERIC_PARSER_ARGS);
+int config_parse_log_level(GENERIC_PARSER_ARGS);
+int config_parse_signal(GENERIC_PARSER_ARGS);
+int config_parse_personality(GENERIC_PARSER_ARGS);
+int config_parse_ifname(GENERIC_PARSER_ARGS);
+int config_parse_ip_port(GENERIC_PARSER_ARGS);
 
 #define DEFINE_CONFIG_PARSE_ENUM(function,name,type,msg)                \
-        int function(const char *unit,                                  \
-                     const char *filename,                              \
-                     unsigned line,                                     \
-                     const char *section,                               \
-                     unsigned section_line,                             \
-                     const char *lvalue,                                \
-                     int ltype,                                         \
-                     const char *rvalue,                                \
-                     void *data,                                        \
-                     void *userdata) {                                  \
-                                                                        \
+        int function(GENERIC_PARSER_ARGS) {                             \
```

Comment:
Hmm, should we err on the side of keeping the argument list visible in this function _definition_?  GENERIC_PARSER_ARGS works nicely in the big list of function declarations, but maybe we shouldn't use it outside of that.

Reply:
The macro is defined just a few lines above, so I think it is OK to also use the macro here.

Reply:
ehh.  I'm not used to reading function definitions that way.

Just having `#undef GENERIC_PARSER_ARGS` somewhere would make me happier.  It communicates the point made in the commit message, that we don't expect people to remember it.  If you think the `#undef` would look ok placed after these two function definitions, then I defer to submitters judgement.

BTW apologies for clicking the wrong button, I was trying to do a full review from the start.

Reply:
It'd need to be undefined *after* `DEFINE_CONFIG_PARSE_ENUM` is used, so there doesn't seem to be a nice way to do this.

Reply:
ohh. sorry.